### PR TITLE
lib: Add api to add remotes from flatpakrepo files

### PR DESCRIPTION
**Pull request by alexlarsson**
This adds flatpak_remote_new_from_file() allowing you to pre-generate a FlatpakRemote for a flatpakrepo file, including new features like titles, filters, etc. It also adds flatpak_installation_add_remote() which you can use instead of modify_remote, in particular with if_needed == TRUE, which handles the reseting any pre-existing filters when adding a non-filtered flatpakref file (like the flathub default one).

@kalev This is for you!

### CODE REVIEW

File: doc/reference/flatpak-sections.txt [hunk 1]
```diff
@@ -151,6 +151,7 @@ flatpak_ref_get_type
 <TITLE>FlatpakRemote</TITLE>
 FlatpakRemote
 flatpak_remote_new
+flatpak_remote_new_for_file
```
Comment (matthiasclasen) [inline doc/reference/flatpak-sections.txt]: the header calls it flatpak_remote_new_from_file

**Comment by matthiasclasen** [comment]
Do we define a new drop dir for these ? Or just reuse /etc/flatpak/remotes.d/ ?

**Comment by alexlarsson** [comment]
I used the same dir name, the extension is different anyway.

**Comment by alexlarsson** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**